### DISCUSSION TITLE
Help Improve Precision for FlickrLogos32 Dataset with YOLO

### DISCUSSION THREAD

**Discussion by Adrianq88** [ultralytics/ultralytics#17983] [Q&A]
Hi everyone,

**I am training YOLO on Google Colab environment, using PRO plan with T4 GPU.**

I've been training on the FlickrLogos32 dataset using the YOLOv9 model, but I'm struggling to achieve a precision higher than 85%. My goal is to reach at least 90% precision. Below are the results from my various training runs:

35 epochs: Precision P: 0.812, mAP50: 0.699
300 epochs: Precision P: 0.852, mAP50: 0.748
750 epochs with patience=100 (best model at epoch 650): Precision P: 0.848, mAP50: 0.736
For all of these trainings, I used the default YOLOv9 small model with the following code:

```
import os
from ultralytics import YOLO

model = YOLO("yolov9s.pt")

results = model.train(data=os.path.join(ROOT_DIR, "flickr_setup.yaml"), epochs=300, save_period=10)
```

After these runs, I started experimenting with data augmentation and adjusted some hyperparameters and I also switched to yolov9 medium, not small anymore. Here's the configuration I tried:

```
results = model.train(
    data=os.path.join("/content/gdrive/My Drive/YOLO/FlickrLogos32", "flickr_setup.yaml"),
    epochs=600,  # More epochs for fine-tuning
    imgsz=640,   # Larger image size
    save_period=5,
    batch=32,     # Increased batch size
    workers=8,
    lr0=0.0025,   # Lower initial learning rate
    cos_lr=True,  # Cosine learning rate schedule
    mosaic=1,
    mixup=0.1,    # Reduced mixup intensity
    erasing=0.3,  # Reduced erasing intensity
    dropout=0.2,  # Increased dropout
    box=5.0,      # Lower box loss weight
    cls=0.5,      # Lower classification weight
    warmup_epochs=5,
    augment=True,
    patience=30,
)
```

However, with these parameters, training stopped at epoch 179, with a validation result of:

BoxP: 0.834

Here are the metrics after training:

![image](https://github.com/user-attachments/assets/15b33e30-56fc-4389-8ca6-e5e820b30375)

**My Question is:**
What can I improve to boost the performance of this training? Is it possible to reach a precision higher than 85% on this dataset, or is 83-85% the best I can expect for FlickrLogos32?

I would really appreciate any advice or suggestions you can provide. Thanks in advance!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Adrianq88, thank you for sharing your detailed training results and experiments with YOLO 🚀! It’s great to see your progress and the different techniques you’re applying to improve performance.

We recommend checking out the [Docs](https://docs.ultralytics.com) for additional insights, including [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) and example configurations that might help refine your approach.

If this is a 🐛 Bug Report (e.g., unexpected training behavior), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) including your dataset and training script, so we can investigate further.

For custom training ❓ questions like this, providing as much detail as possible (which you've already done 👍) is super helpful. Additionally, here are a few suggestions to consider:
- Share a few sample images and annotations from your dataset to ensure there’s no dataset imbalance or label misconfiguration.
- Review the hyperparameter settings in your `.yaml` file and verify they align with recommendations from our [Guides](https://docs.ultralytics.com/guides/).
- Experiment with the larger models in the YOLO family (e.g., `yolov9m.pt` or `yolov9l.pt`) if computational resources allow, as larger models may capture more complex patterns in your data.

## Upgrade

Before proceeding further, please ensure you are running the latest version of `ultralytics` with all dependencies updated, as newer versions could contain performance improvements or fixes that may address your challenges. Here's how to upgrade:

```bash
pip install -U ultralytics
```

## Environments

YOLO works seamlessly in various verified environments. Below are some options you can explore to potentially streamline your experiments:
- **Notebooks** with free GPU resources: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: Refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/).
- **Amazon** Deep Learning AMI: Refer to the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker Image**: See setup in our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, our [Continuous Integration (CI)](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) systems are passing, validating YOLO's functionality across platforms daily.

Lastly, this is an automated response 🙂. An Ultralytics engineer will follow up with additional insights and suggestions to help you optimize your training process. Good luck, and happy training! 💪

**Comment by glenn-jocher** [comment]
Thank you for sharing detailed information about your training process. To improve precision, consider these suggestions: ensure that your dataset is thoroughly labeled and balanced, with diverse and high-quality samples. Experiment with longer training durations and fine-tune learning rates further. Leverage transfer learning by starting with pretrained weights of a larger model variant. Additionally, explore techniques like image tiling for small objects and advanced augmentation strategies. Small improvements in data quality and parameter tuning can collectively impact overall model performance. Let me know how it goes!
